### PR TITLE
Upgrade `snakeyaml` dependency.

### LINKED ISSUE #88: Upgrade `snakeyaml` dependency.
Newer version of `snakeyaml` is available to upgrade.

### PR DESCRIPTION
**Pull request by mashhurs**
## What does this PR change?
Upgrades `snakeyaml` to 2.0.

### How to test locally?
- build jar with `./gradlew clean vendor`
    - creates logstash-filter-useragent-3.3.5.jar
- define a config to use the plugin
```
input {
  stdin {}
}
filter {
  useragent {
    source => 'message'
  }
}
output {
 stdout {}
}
```

- run Logstash and check if plugin is parsing the user agent info
```
Mozilla/5.0 (iPad; U; CPU OS 3_2_1 like Mac OS X; en-us) AppleWebKit/531.21.10 (KHTML, like Gecko) Mobile/7B405
{
          "host" => {
        "hostname" => "Mashhurs-MacBook-Pro.local"
    },
      "@version" => "1",
    "@timestamp" => 2023-09-19T03:44:35.218216Z,
    "user_agent" => {
            "os" => {
               "full" => "iOS 3.2.1",
            "version" => "3.2.1",
               "name" => "iOS"
        },
          "name" => "Mobile Safari UI/WKWebView",
        "device" => {
            "name" => "iPad"
        }
    },
       "message" => "Mozilla/5.0 (iPad; U; CPU OS 3_2_1 like Mac OS X; en-us) AppleWebKit/531.21.10 (KHTML, like Gecko) Mobile/7B405",
         "event" => {
        "original" => "Mozilla/5.0 (iPad; U; CPU OS 3_2_1 like Mac OS X; en-us) AppleWebKit/531.21.10 (KHTML, like Gecko) Mobile/7B405"
    }
}
```

- Closes #88

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -94,7 +94,7 @@ task verifyYaml(type: Verify, dependsOn: [patchYaml, processResources]) {
 
 dependencies {
   implementation group: 'org.apache.commons', name: 'commons-collections4', version: '4.1'
-  implementation group: 'org.yaml', name: 'snakeyaml', version: '1.33'
+  implementation group: 'org.yaml', name: 'snakeyaml', version: '2.0'
```
Comment (andsel) [inline build.gradle:97]: Snakeyaml was updated on Logstash core with https://github.com/elastic/logstash/pull/15125, shipped with `v8.9.0`. Should we constraint this version of the plugin to that LS versions ? (add a dependency on LS 8.9.0+ in gemspec)

**Review by andsel — COMMENTED** [review]
Left a doubt on the fact to restrict LS version compatibility.

File: build.gradle (hunk 1 above)
Comment (mashhurs) [reply build.gradle:97]: With this change, we see with the CIs that it is compatible with the latest 8.x and 7.x version. We should investigate the case where LS core has a lower version than plugin but for now, pessimistic locking without clarification would block some users after release.

**Review by andsel — APPROVED** [review]
LGTM

**Pull request merged**